graylog2-web-interface/src/views/components/sidebar/highlighting/HighlightingColorForm.tsx

```diff
@@ -183,20 +195,31 @@ const createNewColor = (newType: string) => {
   throw new Error(`Invalid color type: ${newType}`);
 };
 
+export const validateColoringType = (value, fieldIsNumeric) => {
+  if (!value || value === '') {
+    return 'Coloring is required';
+  }
+
+  if (!fieldIsNumeric && value === 'gradient') {
+    return 'A gradient can only be defined for numeric fields.';
+  }
+
+  return undefined;
+};
+
 const HighlightingColorForm = ({ field }: Props) => {
   const isNumeric = field?.type?.isNumeric() ?? false;
-  const isDisabled = field === undefined;
 
   const { setFieldValue } = useFormikContext();
 
   const onChangeType = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
     const { target: { value: newType } } = e;
-    setFieldValue('color', createNewColor(newType));
+    setFieldValue('color', createNewColor(newType as ColorType));
   }, [setFieldValue]);
 
   return (
-    <Field name="color.type" validate={_typeIsRequired}>
-      {({ field: { name, value } }, meta) => (
+    <Field name="color.type" validate={(value) => validateColoringType(value, isNumeric)}>
```

Comment:
When changing the selected field and the value of `isNumeric` changes, it takes a second render to provide the correct value for `validateColoringType` and I am not sure why.

Currently it behaved like:
- User selects to a numeric field
- we are rerendering the form
- `isNumeric` is true
- we are validating the coloring type, but inside `validateColoringType` the variable `fieldIsNumeric` is still false